guildwars2/characters.py

```diff
@@ -256,6 +257,229 @@ def handle_duplicates(upgrades):
+            rune_item = await self.fetch_item(rune)
+            bonuses = rune_item["details"]["bonuses"]
+            count = 0
+            for bonus in bonuses:
+                if count < runecount:
+                    pattern_single = re.compile("^\+\d{1,} ")
+                    pattern_all_stats = re.compile(".* [s,S]tats$")
+                    pattern_percentage = re.compile("^\+\d{1,}% ")
+                    # Regex deciding if it's a stat
+                    if pattern_all_stats.match(bonus):
+                        modifier = re.sub(' .*$', '', bonus)
+                        modifier = re.sub('\+', '', modifier)
+                        attr_dict["Power"] += int(modifier)
+                        attr_dict["Vitality"] += int(modifier)
+                        attr_dict["Toughness"] += int(modifier)
+                        attr_dict["Precision"] += int(modifier)
+                        attr_dict["Ferocity"] += int(modifier)
+                        attr_dict["Healing"] += int(modifier)
+                        attr_dict["ConditionDamage"] += int(modifier)
+                    elif pattern_single.match(bonus):
+                        # Regex deciding the attribute name + modifier
+                        modifier = re.sub(' .*$', '', bonus)
+                        modifier = re.sub('\+', '', modifier)
+                        attribute_name = re.sub('^.* ', '', bonus)
+                        if attribute_name in attr_dict:
+                            attr_dict[attribute_name] += int(modifier)
+                    elif pattern_percentage.match(bonus):
+                        modifier = re.sub(' .*$', '', bonus)
+                        modifier = re.sub('\+', '', modifier)
+                        attribute_name = re.sub(' Duration', 'Duration', bonus)
+                        attribute_name = re.sub('^.* ', '', attribute_name)
+                        if attribute_name in attr_dict:
+                            attr_dict[attribute_name] += int(modifier)
+                    # Amount of runes equipped
+                    count += 1
+
+        # Calculate base value depending on char level
+        basevalue = self.calcBaselvl(level, 0, lvl_dict)
+        attr_dict["Power"] += basevalue
+        attr_dict["Vitality"] += basevalue
+        attr_dict["Toughness"] += basevalue
+        attr_dict["Precision"] += basevalue
+        # Mapping for old attribute names
+        attr_dict["Concentration"] += attr_dict["BoonDuration"]
+        attr_dict["Ferocity"] += attr_dict["CritDamage"]
+        attr_dict["Expertise"] += attr_dict["ConditionDuration"]
+        # Calculate derivative attributes
+        # Reset to default after mapped to new attribute name
+        attr_dict["CritDamage"] = 150 + round(attr_dict["Ferocity"] / 15, 2)
+        attr_dict["BoonDuration"] = round(attr_dict["Concentration"] / 15, 2)
+        attr_dict["ConditionDuration"] = round(attr_dict["Expertise"] / 15, 2)
```

Comment:
These result in a float, which when at 0 looks like `0.0`, while other values are simply `0`

Reply:
Added cast to int if 0.0